controller/src/lib.rs

```diff
@@ -632,6 +747,215 @@ impl Controller {
         }
     }
 
+    /// Set the power mode for a set of transceiver modules.
+    ///
+    /// This method may be used regardless of whether a module uses hardware
+    /// control or software override for controlling the power.
+    pub async fn set_power_mode(&self, modules: ModuleId, mode: PowerMode) -> Result<(), Error> {
+        // How we proceed largely depends on two things: whether we're turning
+        // the power OFF entirely, and whether a module has set software
+        // override of the `LPMode` pin.
+        match mode {
+            PowerMode::Off => {
+                self.assert_lpmode(modules).await?;
+                self.assert_reset(modules).await?;
+                self.disable_power(modules).await
+            }
+            other => {
```

Comment:
I'd be tempted to use the weird `@` syntax, to explicitly spell out the power modes captured by this branch:
```suggestion
            other @ PowerMode::Low | PowerMode::High => {
```

Reply:
Never had a reason to use that, but I like it here thanks!